### PR TITLE
[Fixes #8026] Implement support for TMS thumbnail backgrounds

### LINKED ISSUE #8026: Implement support for TMS thumbnail backgrounds
An additional option to the `THUMBNAIL_BACKGROUND.options` dict to enable TMS based services.
The TMS URL will have to use the usual `{z}/{x}/{y}` parameters, but when the `tms` options is set to `True` the `y` will be flipped.

### PR DESCRIPTION
**Pull request by giohappy**
Reference #8026 

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/GeoNode/geonode/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in the documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):
- [x] There is a ticket in https://github.com/GeoNode/geonode/issues describing the issue/improvement/feature (a notable exemption is, changes not visible to end-users)
- [x] The issue connected to the PR must have Labels and Milestone assigned
- [ ] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[Fixes #<issue_number>] Title of the Issue"
- [ ] New unit tests have been added covering the changes, unless there is an explanation on why the tests are not necessary/implemented
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the QA checks: flake8 geonode
- [ ] Commits changing the **settings**, **UI**, **existing user workflows**, or adding **new functionality**, need to include documentation updates
- [ ] Commits adding **new texts** do use gettext and have updated .po / .mo files (without location infos)

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

### CODE REVIEW

File: geonode/thumbs/background.py [hunk 1]
```diff
@@ -182,6 +182,7 @@ def __init__(
         # Slippy Maps specific attributes (to be overwritten in specific background classes)
         self.url = options.get("url", None)
         self.tile_size = options.get("tile_size", 256)
+        self.tms = True if options.get("tms") and options.get("tms") == True else False
```
Comment (afabiani) [inline geonode/thumbs/background.py:185]: Easier and more readable writing this one as:

`self.tms = options.get("tms", False)`

Comment (giohappy) [reply geonode/thumbs/background.py:185]: yes, sure @afabiani. I'm going to fix it. However I would check if the value is a boolean.

Comment (giohappy) [reply geonode/thumbs/background.py:185]: @afabiani it should be like this:

`self.tms = options.get("tms") if options.get("tms") and type(options.get("tms")) is bool else False`

**Review by afabiani — DISMISSED** [review]

**Comment by afabiani** [comment]
There is a pep8 issue

```
^@^@geonode/settings.py:1970:5: E265 block comment should start with '# '
geonode/settings.py:1973:5: E265 block comment should start with '# '
```

**Review by afabiani — APPROVED** [review]

**Pull request merged**